### DISCUSSION TITLE
An error occurred after plugin 'skywalking-client-js' was paired with Axios.

### DISCUSSION THREAD

**Discussion by sevenyby** [apache/skywalking#8723] [Q&A]
Environment: Window 11, Chrome 99.0.4844.74,

SkyWalking Plugins:  skywalking-client-js 0.7.0, 

App: axios 0.21.4, nodejs v16.14.0, vue 3.2.31

--------------------------------------------------------------------------------------------------------------------------------------------------

Problems : Uncaught TypeError: Cannot read properties of undefined (reading 'apply')

![image](https://user-images.githubusercontent.com/9942770/159387473-bf699b01-3e15-4491-a6e1-0be3e8f42e73.png)

Relevant code:

router.js:
![image](https://user-images.githubusercontent.com/9942770/159387814-176bc680-16ef-4510-878c-b3966753b24b.png)

vue.config.js:
![image](https://user-images.githubusercontent.com/9942770/159387960-4ba2d5ce-2dec-492e-bddc-ef8d625fe30b.png)

http-demo.js(Common js of axios):
![image](https://user-images.githubusercontent.com/9942770/159388022-308d32dc-18d2-4b8e-a05f-9ee70ed19fc2.png)

DemoService.js:
![image](https://user-images.githubusercontent.com/9942770/159388168-1180b7bf-2d38-4b19-a4c1-a53498bf74ea.png)

DemoCustomer.vue:
![image](https://user-images.githubusercontent.com/9942770/159388690-3ca045e0-1eba-4c9d-8adc-c44a279baa42.png)

Looking forward to your help in solving this problem. Thx.

**Comment by wu-sheng** [comment]
FYI @arugal @Fine0830

**Comment by Fine0830** [comment]
You need to register client-js in your main.ts. @sevenyby

**Reply by sevenyby** [reply]
I moved the registered content to 'main.js', And I got the same error again.

console log:

![image](https://user-images.githubusercontent.com/9942770/159422668-89e78876-2493-4eb6-98e5-774a9eedf86c.png)

main,js

![image](https://user-images.githubusercontent.com/9942770/159422795-2a3ece6b-31e8-44ec-a9ea-6d6e6d1fd881.png)

I can also provide anything that can help you troubleshoot the problem.

**Reply by Fine0830** [reply]
I can't reproduce this. Here is my [demo](https://github.com/Fine0830/skywalking-rocketbot-ui/tree/feat/browser) , it works well.

**Comment by warrmr** [comment]
I have this issue also but am unable to provide further information as to how/where it may occur so have refrained from creating a github issue just in case its "programmer error" < me. 

It appears that if our app reports an error (very infrequently) Skywalking attempts to send this to the opm server but fails and when it fails to do so you either get this error or an unable to convert circular structure to json error. 

I am using Vue also and have the following sections in the main and router js files. I am running the latest skywalking-client-js for the fix with the FMP time issue. Thanks @Fine0830 for sorting that so quickly appreciate it :) 

Do I still need to use the ClientMonitor.register method if I am using the ClientMonitor.setPerformance method in the router? @Fine0830 

package.json
```
"skywalking-client-js": "^0.8.0",
```

main.js 
```
Vue.config.errorHandler = (error) => {
    ClientMonitor.reportFrameErrors(
        {
            collector: "https://skywalking",
            service: name,
            pagePath: router.currentRoute,
            serviceVersion: `v${version}`,
            vue: true,
        },
        error,
    );
};
```

router.js
```router.beforeEach(async (to, _from, next) => {
    ClientMonitor.setPerformance({
        collector: "https://skywalking",
        service: name,
        pagePath: to.path,
        serviceVersion: `v${version}`,
        useFmp: true,
        vue: true,
    });

    return next();
});
```

**Reply by warrmr** [reply]
This issue makes our app look amazing as there is nothing showing in the error dashboard :) However we know it glitches every now and then so expect to see something in there :( 

![image](https://user-images.githubusercontent.com/393693/163281416-5e378ab4-5643-4a20-a57b-b3d185f08d92.png)

**Reply by warrmr** [reply]
One last post before I log off for the evening :) 

Not sure if its related but there seems to be nothing coming into the trace section of the dashboard either from the vue apps.  any ideas?
![image](https://user-images.githubusercontent.com/393693/163281593-00cf491e-a53d-4223-95bf-44aadf6b2613.png)

**Comment by czxin788** [comment]
my error is :
![image](https://user-images.githubusercontent.com/13861904/164954213-67a19828-a516-49bf-ad83-3b923929ce42.png)
and the segments is empty:
![image](https://user-images.githubusercontent.com/13861904/164954293-3c08751c-7c82-4451-8f21-8cdc28897812.png)
